source/ai-server-logger.js

```diff
@@ -0,0 +1,122 @@
+(function() {
+    'use strict';
+
+    angular.module('ai.public.serverLogger', ['ng'])
+        .config(function($provide) {
+            $provide.decorator('$log', function($delegate, $injector) {
+                var logger = $injector.get('AiServerLogger');
+                var delegateError = $delegate.error;
+                var delegateWarn = $delegate.warn;
+                var delegateInfo = $delegate.info;
+                var delegateDebug = $delegate.debug;
+
+                var enhanceLoggingFunction = function(loggingFunction, severity) {
+                    return function() {
+                        var args = [].slice.call(arguments);
```

Comment:
It seems like apply can handle array like objects just fine.. http://stackoverflow.com/a/20375156/689411